crates/bindings-csharp/BSATN.Runtime/Repr.cs

```diff
@@ -1,27 +0,0 @@
-namespace SpacetimeDB.Internal;
-
-using System.Runtime.InteropServices;
-
-// We store time information in microseconds in internal usages.
-//
-// These utils allow to encode it as such in FFI and BSATN contexts
-// and convert to standard C# types.
-
-[StructLayout(LayoutKind.Sequential)] // we should be able to use it in FFI
-[SpacetimeDB.Type] // we should be able to encode it to BSATN too
-public partial struct DateTimeOffsetRepr(DateTimeOffset time)
```

Comment:
It's a matter of taste, of course, but I kind of liked the intermediate representation as a separate type you can convert from, especially since `struct`s are cheap / zero-copy wrappers.

Reply:
I had found a server-side bsatn parsing issue in blackholio and was wondering if this was related. Turned out not to be, the actual fix is here: https://github.com/clockworklabs/SpacetimeDB/pull/2164/files#diff-8b70d4eeef215ef2f1e1a16e36b20ae9a943e1a3aeb823c260fc056842b570a8R158 . I can revert this. I need to implement a custom Timestamp type per phoebe's proposal anyway which this can be used for.

Reply:
It's fine either way. As you said, it's going to change soon anyway.

Reply:
Ah, and you'd already fixed that, I missed it. Welp!